content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,228 +11,193 @@ weight: 10
 
-### azureFile {#azurefile}
+### Storage plugins
 
-The `azureFile` volume type mounts a Microsoft Azure File volume (SMB 2.1 and 3.0)
-into a pod.
+A [persistentVolumeClaim](#persistentvolumeclaim) is a declarative way to ask for storage, without preference for the underlying implementation. 
 
-For more details, see the [`azureFile` volume plugin](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/staging/volumes/azure_file/README.md).
+This is powered by storage plugins. More specifically, Kubernetes implements the {{< glossary_tooltip text="Container Storage Interface" term_id="csi" >}} (CSI),
+which lets developers implement storage plugins without needing to make changes to the code of
+Kubernetes itself. Visit
+[Kubernetes Container Storage Interface Documentation])(https://kubernetes-csi.github.io/docs/)
+to learn more about CSI. 
 
-#### azureFile CSI migration
+[FlexVolume](#flexvolume) is an another mechanism for storage plugins. Unlike CSI, FlexVolume is
+more closely tied to Kubernetes and its storage implementation. FlexVolume was developed before CSI.
 
-{{< feature-state for_k8s_version="v1.21" state="beta" >}}
+Storage plugin interfaces:
 
-The `CSIMigration` feature for `azureFile`, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `file.csi.azure.com` Container
-Storage Interface (CSI) Driver. In order to use this feature, the [Azure File CSI
-Driver](https://github.com/kubernetes-sigs/azurefile-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationAzureFile`
-[feature gates](/docs/reference/command-line-tools-reference/feature-gates/) must be enabled.
+   * [CSI](#csi)
+   * [FlexVolume](#flexvolume)
+   * [persistentVolumeClaim](#persistentvolumeclaim)
 
-Azure File CSI driver does not support using same volume with different fsgroups, if Azurefile CSI migration is enabled, using same volume with different fsgroups won't be supported at all.
 
-### cephfs
+### External Volume Types
 
-A `cephfs` volume allows an existing CephFS volume to be
-mounted into your Pod. Unlike `emptyDir`, which is erased when a pod is
-removed, the contents of a `cephfs` volume are preserved and the volume is merely
-unmounted. This means that a `cephfs` volume can be pre-populated with data, and
-that data can be shared between pods. The `cephfs` volume can be mounted by multiple
-writers simultaneously.
+Kubernetes directly implements some volume types for communicating with externally managed volumes, such as AWS EBS or Azure Disk Storage. These volume types are described as in-tree storage plugins. 
 
-{{< note >}}
-You must have your own Ceph server running with the share exported before you can use it.
-{{< /note >}}
+The CSI was introduced to standardize connectivity to externally managed volumes, such as cloud platform storage. However, Kubernetes still includes many volume types for connecting to specific cloud storage providers. 
```

Comment:
Reword something to something like "Using in-tree plugins are not recommended anymore. All new storage features are only supported for CSI volumes, and in-tree plugins are being replaced by their equivalent CSI drivers".

Reply:
added a callout on this